keras_cv/layers/preprocessing/random_saturation_test.py

```diff
@@ -93,3 +172,29 @@ def test_config(self):
         self.assertTrue(isinstance(config["factor"], core.UniformFactorSampler))
         self.assertEqual(config["factor"].get_config()["lower"], 0.3)
         self.assertEqual(config["factor"].get_config()["upper"], 0.8)
+
+    def test_correctness_with_tf_adjust_saturation_normalized_range(self):
```

Comment:
@LukeWood I really like the pattern of A/B testing this before merging these changes, but I'm not sure it belongs in unit tests permanently. wdyt?

Reply:
Hi @ianstenbit 
I'm also wondering if there has better solution. I think the maintenance of old layers will be a problem if
1. `BaseImageAugmentationLayer` is modified or deprecated
2. It is implicit to have a class definition in unit tests

If the consistency of outputs is the major concern, maybe we can extract the computation part (`tf.multiply`, `tf.image`, ..., etc) instead of whole class definition for unit tests?

Reply:
> @LukeWood I really like the pattern of A/B testing this before merging these changes, but I'm not sure it belongs in unit tests permanently. wdyt?

So, I see why it might not belong in unit tests permanently - but we do need a way to monitor the deltas here during code reviews.  We can't really just ask contributors "did you test this" - we could ask for a colab perhaps, but I think it fits a bit more cleanly in the unit tests.

Personally, I'd opt for include in unit tests for now, then we can remove them all at the end once we feel we have achieved stability.

Reply:
Okay that sgtm. Thank you Luke and thank you @james77777778 for your work on this!

Reply:
(One other option we could consider is a/b testing in the benchmark script).

Personally I find that a little bit nicer as then we don't have to have two copies of `OldRandomSaturation` wdyt?

Reply:
ok yeah I agree with @ianstenbit - let's put this in the benchmarks.

Reply:
Got it. I'm going to move a/b testing into `benchmarks/vectorized_random_saturation.py`.

Just one question:
During the migration of vectorizing preprocessing layers, the number of benchmark script might increase rapidly. Should I make new folder to put it?